src/middlewared/middlewared/pytest/unit/common/smart/test_smartctl.py

```diff
@@ -10,19 +10,28 @@
 
 @pytest.mark.asyncio
 async def test__get_smartctl_args__disk_nonexistent():
-    assert await get_smartctl_args(None, {}, "ada0") is None
+    with patch("middlewared.common.smart.smartctl.IS_LINUX", Mock(return_value=False)):
```

Comment:
You are mocking it with callable, while it is a scalar value. `bool(Mock(return_value=False))` is `True` so your tests are behaving like we are on Linux. Have you ran them, are they passing? (they should not, if they are passing, it is a ringing bell)